### PR TITLE
[Build]: Support reproducible build for release branches

### PR DESCRIPTION
**Pull request by xumia**
#### Why I did it
Support reproducible build for release branches

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [x] 202106
- [x] 202111

### CODE REVIEW

**Review by liushilongbuaa — APPROVED** [review]

File: .azure-pipelines/azure-pipelines-UpgrateVersion.yml [hunk 1]
```diff
@@ -12,6 +12,8 @@ schedules:
   branches:
     include:
     - 202012
+    - '202[1-9]*'
```
Comment (qiluo-msft) [inline .azure-pipelines/azure-pipelines-UpgrateVersion.yml:15]: The 5th and 6th digits are the month. Let's pattern match them and keep string length fixed.

File: .azure-pipelines/azure-pipelines-repd-build-variables.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+variables:
+  ${{ if and(ge(variables['Build.SourceBranchName'], '202012'), le(variables['Build.SourceBranchName'], '299999')) }}:
+    VERSION_CONTROL_OPTIONS: 'SONIC_VERSION_CONTROL_COMPONENTS=deb,py2,py3,web,git,docker'
+  ${{ if or(lt(variables['Build.SourceBranchName'], '202012'), gt(variables['Build.SourceBranchName'], '299999')) }}:
```
Comment (qiluo-msft) [inline .azure-pipelines/azure-pipelines-repd-build-variables.yml:4]: What happen if the SourceBranchName is not a number? Should we explicitly handle this case?

Comment (xumia) [reply .azure-pipelines/azure-pipelines-repd-build-variables.yml:4]: @qiluo-msft , we use string comparing, will skip the reproducible build feature.

File: .azure-pipelines/azure-pipelines-UpgrateVersion.yml (hunk 1 above)
Comment (xumia) [reply .azure-pipelines/azure-pipelines-UpgrateVersion.yml]: The regex is not supported in AZP yet, it only supports ? and *.

File: .azure-pipelines/azure-pipelines-UpgrateVersion.yml [hunk 3]
```diff
@@ -11,7 +11,11 @@ schedules:
   displayName: Daily Build
   branches:
     include:
-    - 202012
+    - 20????
+    exclude:
+    - 200*
```
Comment (qiluo-msft) [inline .azure-pipelines/azure-pipelines-UpgrateVersion.yml:16]: >200* [](http://example.com/codeflow?start=6&length=4)

also keep excluding pattern fixed length like `200???`

File: .azure-pipelines/azure-pipelines-repd-build-variables.yml (hunk 2 above)
Comment (qiluo-msft) [reply .azure-pipelines/azure-pipelines-repd-build-variables.yml:4]: `VERSION_CONTROL_OPTIONS` is not defined in that case. Is it going to be a problem?

File: .github/workflows/label.yml [hunk 4]
```diff
@@ -13,6 +13,8 @@ on:
     - reopened
     branches:
     - '202012'
+    - '202[1-9]*'
```
Comment (qiluo-msft) [inline .github/workflows/label.yml:16]: >202[1-9]* [](http://example.com/codeflow?start=7&length=9)

limit the length of the branch name?

File: .azure-pipelines/azure-pipelines-repd-build-variables.yml (hunk 2 above)
Comment (xumia) [reply .azure-pipelines/azure-pipelines-repd-build-variables.yml:4]: It is depended how to use the variable.
```
Micro: $(VERSION_CONTROL_OPTIONS)   --- not work
Template: ${{ variables.VERSION_CONTROL_OPTIONS }}  --- empty
Runtime: $[variables.VERSION_CONTROL_OPTIONS ]  --- empty
```
Currently, we are good, we use template way.

File: .azure-pipelines/azure-pipelines-UpgrateVersion.yml (hunk 3 above)
Comment (xumia) [reply .azure-pipelines/azure-pipelines-UpgrateVersion.yml]: Fixed, thanks.

File: .github/workflows/label.yml (hunk 4 above)
Comment (xumia) [reply .github/workflows/label.yml]: Fixed, thanks.

File: .azure-pipelines/azure-pipelines-UpgrateVersion.yml (hunk 1 above)
Comment (xumia) [reply .azure-pipelines/azure-pipelines-UpgrateVersion.yml]: Fixed.

**Review by qiluo-msft — APPROVED** [review]

**Comment by qiluo-msft** [comment]
@xumia Do you need this commit on 202012 branch?

**Pull request merged**